crates/aptos-compression/src/lib.rs

```diff
@@ -108,6 +129,36 @@ pub fn decompress(
     Ok(raw_data)
 }
 
+/// Derived from lz4-rs crate, which starts the compressed payload with the original data size as i32
+/// see: https://github.com/10XGenomics/lz4-rs/blob/0abc0a52af1f6010f9a57640b1dc8eb8d2d697aa/src/block/mod.rs#L162
+fn get_decompressed_size(src: &CompressedData, max_size: usize) -> std::io::Result<usize> {
+    if src.len() < 4 {
+        return Err(Error::new(
+            ErrorKind::InvalidInput,
+            "Source buffer must at least contain size prefix.",
+        ));
+    }
+
+    let size =
```

Comment:
I'm surprised the crate does it like this. I'd have assumed there was an easier way? e.g., converting to a usize from bytes 🤔